src/crypto/openssl/aead.rs

```diff
@@ -190,11 +190,14 @@ impl CipherSuite {
         let mut signer = openssl::sign::Signer::new(openssl::hash::MessageDigest::sha256(), &key)?;
 
         // for current platforms there is no issue casting from usize to u64
-        let aad_len = (aad.len() as u64).to_be_bytes();
-        let ct_len = (encrypted.len() as u64).to_be_bytes();
-        for buf in [&aad_len, &ct_len, nonce, aad, encrypted] {
+        let aad_len = &(aad.len() as u64).to_be_bytes();
+        let ct_len = &(encrypted.len() as u64).to_be_bytes();
+        let tag_len = &(self.auth_tag_len as u64).to_be_bytes();
```

Comment:
is there a `.as_bytes()` equivalent?

Reply:
Unfortunately this is not possible as it not agnostic on which system you are compiling (Big Endian vs. Little Endian).

So what `to_be_bytes()`  does is to call  `self.to_be().to_ne_bytes()` 
where 
```
        #[inline]
        pub const fn to_ne_bytes(self) -> [u8; mem::size_of::<Self>()] {
            // SAFETY: integers are plain old datatypes so we can always transmute them to
            // arrays of bytes
            unsafe { mem::transmute(self) }
        }

        pub const fn to_be(self) -> Self { // or not to be?
            #[cfg(target_endian = "big")]
            {
                self
            }
            #[cfg(not(target_endian = "big"))]
            {
                self.swap_bytes()
            }
        }
```
As you see, the original value is implicitly copied, where the copy is then consumed as there might need to be some byte swaping. Due to the consuming it is a `to` operation.